backend/infrahub/core/branch/models.py

```diff
@@ -553,4 +560,344 @@ async def rebase_graph(self, db: InfrahubDatabase, at: Optional[Timestamp] = Non
         await delete_query.execute(db=db)
 
 
+class FieldMetadata(BaseModel):
+    updated_at: datetime | None = None
+    updated_by: str | None = None
+
+
+class NameValueField(FieldMetadata):
+    value: str = Field(
+        max_length=250, min_length=3, description="Name of the branch (git ref standard)", validate_default=True
+    )
+
+
+class DescriptionValueField(FieldMetadata):
+    value: str = ""
+
+
+class StatusValueField(FieldMetadata):
+    value: BranchStatus = BranchStatus.OPEN
+
+
+class OriginBranchValueField(FieldMetadata):
+    value: str = "main"
```

Comment:
In 99% of the time this would default to "main", but it is still possible to setup Infrahub to use another name for the default branch during first initialization, as such this should probably be a Field with a default_factory which would return `registry.default_branch` when called (not sure if that might lead to some circular imports).